frontend/src/pages/instance/Settings/General.vue

```diff
@@ -78,21 +129,75 @@ export default {
                 projectTypeName: '',
                 stackDescription: '',
                 templateName: '',
-                haConfig: {}
+                haConfig: {},
+                customHostname: ''
             },
             original: {
-                projectName: ''
-            }
+                projectName: '',
+                customHostname: ''
+            },
+            changed: {
+                customHostname: false
+            },
+            errors: {
+                customHostname: ''
+            },
+            url: '',
+            checkingDomain: false,
+            domainStatusValid: false
         }
     },
     computed: {
-        ...mapState('account', ['features']),
+        ...mapState('account', ['features', 'team']),
         isHA () {
             return !!this.instance?.ha
+        },
+        customHostnameValid () {
+            return this.errors.customHostname === '' && this.original.customHostname !== this.input.customHostname
+        },
+        customHostnameAvailable () {
+            const available = this.features.customHostnames
+            return available
+        },
+        customHostnameTeamAvailable () {
+            const available = this.features.customHostnames && this.team.type.properties.features?.customHostnames
+            return available
+        },
+        customHostnameLauncherVersion () {
+            const launcherVersion = this.instance?.meta?.versions?.launcher
+            if (!launcherVersion) {
+                // Not sure the launcher version - could be suspended/started
+                // Be optimistic
+                return true
+            }
+
+            // needs to be  v2.5.0 or better
+            return SemVer.satisfies(SemVer.coerce(launcherVersion), '>=2.4.0')
```

Comment:
```suggestion
            return SemVer.satisfies(SemVer.coerce(launcherVersion), '>=2.5.0')
```

Again bumping this to match what it will ship in.